examples/simple_xo/src/main.rs

```diff
@@ -0,0 +1,246 @@
+use sawtooth_xo::handler::XoTransactionHandler;
+use transact::context::manager::sync::ContextManager;
+use transact::database::{btree::BTreeDatabase, Database};
+use transact::execution::{adapter::static_adapter::StaticExecutionAdapter, executor::Executor};
+use transact::protocol::receipt::TransactionReceipt;
+use transact::protocol::{
+    batch::{BatchBuilder, BatchPair},
+    receipt::StateChange,
+    transaction::{HashMethod, TransactionBuilder},
+};
+use transact::sawtooth::SawtoothToTransactHandlerAdapter;
+use transact::scheduler::{
+    serial::SerialScheduler, BatchExecutionResult, Scheduler, TransactionExecutionResult,
+};
+use transact::signing::{hash::HashSigner, Signer};
+use transact::state::merkle::{self, MerkleRadixTree, MerkleState};
+use transact::state::StateChange as ChangeSet;
+use transact::state::Write;
+
+use hex;
+use sha2::{Digest, Sha512};
+use std::io;
+use std::str;
+use std::sync::{Arc, Mutex};
+
+fn main() {
+    let db = Box::new(BTreeDatabase::new(&merkle::INDEXES));
+    let merkle_db = MerkleRadixTree::new(db.clone(), None).unwrap();
+    let merkle_state = MerkleState::new(db.clone());
+    let orig_root = merkle_db.get_merkle_root();
+
+    let signer = HashSigner::new(vec![00u8, 01, 02]);
+    let current_result = play_game(&orig_root, db.clone(), &signer, "my_game,create,");
+    let (game_address, value) = get_state_change(current_result);
+
+    let state_change = ChangeSet::Set {
+        key: game_address.clone(),
+        value: value.clone(),
+    };
+    let mut state_root = merkle_state.commit(&orig_root, &[state_change]).unwrap();
+    assert_ne!(orig_root, state_root);
+    print_current_state(&value);
+
+    loop {
+        let next_tx = get_next_tx();
+        let current_result = play_game(
+            &state_root,
+            db.clone(),
+            &signer,
+            &format!("my_game,take,{}", next_tx),
+        );
+        let (key, value) = get_state_change(current_result);
+        let state_change = ChangeSet::Set {
+            key: key,
+            value: value.clone(),
+        };
+        state_root = merkle_state.commit(&state_root, &[state_change]).unwrap();
+
+        let value = print_current_state(&value);
+        if value.contains("WIN") || value.contains("TIE") {
+            break;
+        }
+    }
+}
+
+fn print_current_state(value: &Vec<u8>) -> &str {
+    let val = match str::from_utf8(&value) {
+        Ok(v) => v,
+        Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+    };
+
+    let split: Vec<&str> = val.split(",").collect();
```

Comment:
A future enhancement (maybe after this PR?) might be create a Board struct and implement Display for it, and then do Board::from(&str) and println!("{}", board). Same with Game actually.

Reply:
I like that! I will work on that